osquery/tables/system/darwin/homebrew_packages.cpp

```diff
@@ -62,6 +75,246 @@ std::vector<std::string> getHomebrewVersionsFromInfoPlistPath(
   return results;
 }
 
+// Looking for a "auto_updates true" line in a .rb file
+bool checkAutoUpdatesInRubyFile(const std::string& file_path) {
+  std::ifstream file(file_path);
+  if (!file.is_open()) {
+    TLOG << "Error opening file " << file_path;
+    return false;
+  }
+
+  std::string line;
+  std::regex auto_updates_regex(R"(^\s*auto_updates\s*true\s*$)");
+  while (std::getline(file, line)) {
+    if (std::regex_match(line, auto_updates_regex)) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+// Looking for the value of "auto_updates" in a .json file
+// Defaults to false if the key is not found
+bool getBooleanValueFromJsonFile(const std::string& file_path,
+                                 const std::string& key = "auto_updates") {
+  std::string content;
+  auto status = osquery::readFile(file_path, content);
+  if (!status.ok()) {
+    TLOG << "Error reading file " << file_path << ": " << status.toString();
+    return false;
+  }
+
+  pt::ptree tree;
+  std::istringstream is(content);
+  try {
+    pt::read_json(is, tree);
+  } catch (const pt::json_parser_error& e) {
+    TLOG << "Error parsing JSON from file " << file_path << ": " << e.what();
+    return false;
+  }
+
+  return tree.get<bool>(key, false);
+}
+
+// Looking for the app name in a .json file. It is nested under the "artifacts"
+// key, which is an array. One of those arrays has an "app" key, which is an
+// array of strings... and the first string is the app name
+std::string getAppNameFromJsonManifest(const std::string& file_path) {
+  std::string content;
+  auto status = osquery::readFile(file_path, content);
+  if (!status.ok()) {
+    TLOG << "Error reading file " << file_path << ": " << status.toString();
+    return "";
+  }
+
+  pt::ptree tree;
+  std::istringstream is(content);
+  try {
+    pt::read_json(is, tree);
+  } catch (const pt::json_parser_error& e) {
+    TLOG << "Error parsing JSON from file " << file_path << ": " << e.what();
+    return "";
+  }
+
+  // Parsing JSON with arrays is a bit tricky, as the ptree is a tree-like
+  // structure, and we need to iterate over the children of the root node
+  // to find the app name
+
+  auto artifacts = tree.get_child_optional("artifacts");
+  if (!artifacts) {
+    return "";
+  }
+
+  BOOST_FOREACH (auto artifact, artifacts.get()) {
+    if (artifact.second.get_optional<std::string>("app")) {
+      BOOST_FOREACH (auto app, artifact.second.get_child("app")) {
+        return app.second.get_value("");
+      }
+    }
+  }
```

Comment:
Unless I've missed something about why we need the boost foreach, please prefer the c++11 style.

```suggestion
  for (const auto& artifact : artifacts.get()) {
    if (artifact.second.get_optional<std::string>("app")) {
      for (const auto& app : artifact.second.get_child("app")) {
        return app.second.get_value("");
      }
    }
  }
```

Reply:
You didn't miss anything, it's just that my C++ skills have... room for improvement :)